srcs/app/http/response/HttpBaseResponse.cpp

```diff
@@ -37,21 +37,14 @@ void	HttpBaseResponse::Serve_(File file) {
 	SetRawResponse_(200, headers, body);
 }
 
-void	HttpBaseResponse::ExecuteCGI_(File file) {
+void	HttpBaseResponse::ExecuteCGI_(const File &file) {
```

Comment:
I get what these changes are trying to accomplish, solve the race condition between `IsExecutable()` and `execveWr()` and output a proper `HTTP` response.

There is one thing to consider: because there is no `wait()`/`waitpid()` anymore, there is no way to know if `execve()` was successful or not, and in this case, the response is `200 OK` with `500 Internal Server Error` set as the body message.

Reply:
I'm experimenting with some idea I have.
It's not just the race condition.
You can do a 
```
$ touch abc
$ chmod +x abc
```
and then add that abc file as a binary handler. The server will have effectively encountered a internal server error, but it won't report it.

Reply:
I think you can absolutely know if execve was successful without a blocking waitpid. [Here](url)

```
void sigchld_handler(int signum)
{
    pid_t pid;
    int   status;
    while ((pid = waitpid(-1, &status, WNOHANG)) != -1)
    {
        unregister_child(pid, status);   // Or whatever you need to do with the PID
    }
}
...
signal(SIGCHILD, sigchld_handler);
...
```